addons/sourcemod/scripting/gokz-replays/recording.sp

```diff
@@ -65,6 +86,35 @@ void OnClientAuthorized_Recording(int client)
 	}
 }
 
+void OnClientDisconnect_Recording(int client)
+{
+	// Stop exceptions if OnClientPutInServer was never ran for this client id.
+	// As long as the arrays aren't null we'll be fine.
+	if (runningTimers[client] == null)
+	{
+		return;
+	}
+
+	// Trigger all timers early
+	if(!IsFakeClient(client))
+	{
+		// We have to clone the array because the timer callback removes the timer
+		// from the array we're running over, and doing weird tricks is scary.
+		ArrayList timers = runningTimers[client].Clone();
+		for (int i = 0; i < timers.Length; i++)
+		{
+			Handle timer = timers.Get(i);
+			TriggerTimer(timer, false);
+		}
+		delete timers;
+	}
+
+	recordedRecentData[client].Clear();
+	recordedRunData[client].Clear();
+	recordedPostRunData[client].Clear();
+	runningTimers[client].Clear();
```

Comment:
You're doing these 4 lines (112-115) twice, here and in OnClientPutInServer, why? Strikes me as unnecessary

Reply:
I would imagine you also need to reset the postRunRecording field here.

Reply:
> You're doing these 4 lines (112-115) twice, here and in OnClientPutInServer, why? Strikes me as unnecessary

You're right, it is completely unnecessary. I just put it there for completeness: I intend to reset everything, so I reset everything.
The clear call in OnClientDisconnect is only there to reduce memory usage.

> I would imagine you also need to reset the postRunRecording field here.

It is already being reset on line 49.

Reply:
I believe his point was to either clear all of them here as well, including `postRunRecording`, or only do it in `OnClientPutInServer` for all of them. Just seems like a kinda random decision on what to clear and what not to. I do like clearing them in both for memory reasons tho.

Reply:
Doing them in both is okay, I suppose. Better safe than sorry, but to avoid duplication consider extracting them into a method or something. That might be overkill for something that only appears twice but I think there's enough lines to justify it. Also better for readability and expanding on in future.

Reply:
You're right, I abstracted away the clearing into it's own function and just call that instead.